nav2_theta_star_planner/include/nav2_theta_star_planner/theta_star_planner.hpp

```diff
@@ -0,0 +1,89 @@
+// Copyright 2020 Anshumaan Singh
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_THETA_STAR_PLANNER__THETA_STAR_PLANNER_HPP_
+#define NAV2_THETA_STAR_PLANNER__THETA_STAR_PLANNER_HPP_
+
+#include <iostream>
+#include <cmath>
+#include <string>
+#include <chrono>
+#include <queue>
+#include <algorithm>
+#include <memory>
+#include <vector>
+#include "rclcpp/rclcpp.hpp"
+#include "nav2_core/global_planner.hpp"
+#include "nav_msgs/msg/path.hpp"
+#include "nav2_util/robot_utils.hpp"
+#include "nav2_util/lifecycle_node.hpp"
+#include "nav2_costmap_2d/costmap_2d_ros.hpp"
+#include "nav2_costmap_2d/cost_values.hpp"
+#include "nav2_util/node_utils.hpp"
+#include "nav2_theta_star_planner/theta_star.hpp"
+
+namespace nav2_theta_star_planner
+{
+
+class ThetaStarPlanner : public nav2_core::GlobalPlanner
+{
+public:
+  void configure(
+    const rclcpp_lifecycle::LifecycleNode::WeakPtr & parent,
+    std::string name, std::shared_ptr<tf2_ros::Buffer> tf,
+    std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmap_ros) override;
+
+  void cleanup() override;
+
+  void activate() override;
+
+  void deactivate() override;
+
+  nav_msgs::msg::Path createPlan(
+    const geometry_msgs::msg::PoseStamped & start,
+    const geometry_msgs::msg::PoseStamped & goal) override;
+
+protected:
+  std::shared_ptr<tf2_ros::Buffer> tf_;
+  rclcpp::Clock::SharedPtr clock_;
+  rclcpp::Logger logger_{rclcpp::get_logger("SmacPlanner2D")};
```

Comment:
Smac Planner? :-)

Reply:
It wasn't working for some reason, when I typed it out. I was having issues with the rebase installation (it installed once without complaints, but since then has been throwing up issues), so I had to put that segment as it is, forgot to change this part